src/PlaneWaveBasis.jl

```diff
@@ -591,25 +591,21 @@ end
 reinterpret_real(x) = reinterpret(real(eltype(x)), x)
 reinterpret_complex(x) = reinterpret(Complex{eltype(x)}, x)
 
-function pack_ψ(basis::PlaneWaveBasis, ψ)
+function pack_ψ(ψ)
     # TODO as an optimization, do that lazily? See LazyArrays
     vcat([vec(ψk) for ψk in ψ]...)
 end
 
-function unpack_ψ(basis::PlaneWaveBasis{T}, x) where T
-    model = basis.model
-    filled_occ = filled_occupation(model)
-    n_spin = model.n_spin_components
-    n_bands = div(model.n_electrons, n_spin * filled_occ)
-
-    lengths = length.(G_vectors.(basis.kpoints)) .* n_bands
+function unpack_ψ(x, ψ_for_shape) where T
```

Comment:
I think I would pass the shape explicitly or even better do an in-place version of this function. This interface is a bit unusual.